pandas/tests/test_generic.py

```diff
@@ -824,10 +824,27 @@ def test_interp_scipy_basic(self):
 
     def test_interp_limit(self):
         s = Series([1, 3, np.nan, np.nan, np.nan, 11])
+
         expected = Series([1., 3., 5., 7., np.nan, 11.])
         result = s.interpolate(method='linear', limit=2)
         assert_series_equal(result, expected)
 
+        # Provide 'forward' (the default) explicitly here.
+        result = s.interpolate(method='linear', limit=2,
+                               limit_direction='forward')
+        assert_series_equal(result, expected)
+
+        # These tests are for issues #9218 and #10420
+        expected = Series([1., 3., 5., np.nan, 9., 11.])
+        result = s.interpolate(method='linear', limit=1,
+                               limit_direction='symmetric')
+        assert_series_equal(result, expected)
+
+        expected = Series([1., 3., np.nan, 7., 9., 11.])
```

Comment:
actually we don't have a full tests for 10420. E.g. I think ``Series([np.nan.....])`` is the idea